pandas/core/base.py

```diff
@@ -289,6 +289,42 @@ def _try_aggregate_string_function(self, arg: str, *args, **kwargs):
             f"'{arg}' is not a valid function for '{type(self).__name__}' object"
         )
 
+    def _reconstruct_func(self, func, *args, **kwargs):
```

Comment:
ok this is adding a lot of code where we already do a lot of this in groupby. I would start by moving 

```
        from pandas.core.groupby.generic import (
            _is_multi_agg_with_relabel,
            _maybe_mangle_lambdas,
            _normalize_keyword_aggregation,
        )
```
all of that to a new module: pandas.core.aggregation (change the current code to use it).

then add reconstruct_func there.

de-privatize all of these (remove the leading underscores).

Reply:
this function needs some work, but would like to see a reorg first. (the moving can be a separate pre-cursor PR if its easier)

Reply:
thanks for your comment! @jreback 

I will do the code change following your suggestions above. and how about doing reorg and making everything work first and collecting reviews from maintainers, and then I will split the reorg into a separate PR?